packages/storage/src/prefixed_storage.rs

```diff
@@ -88,21 +90,31 @@ impl<'a> ReadonlyPrefixedStorage<'a> {
             prefix: to_length_prefixed_nested(namespaces),
         }
     }
+}
 
-    pub fn get(&self, key: &[u8]) -> Option<Vec<u8>> {
+impl Storage for ReadonlyPrefixedStorage<'_> {
+    fn get(&self, key: &[u8]) -> Option<Vec<u8>> {
         get_with_prefix(self.storage, &self.prefix, key)
     }
 
     #[cfg(feature = "iterator")]
     /// range allows iteration over a set of keys, either forwards or backwards
-    pub fn range<'b>(
+    fn range<'b>(
         &'b self,
         start: Option<&[u8]>,
         end: Option<&[u8]>,
         order: Order,
     ) -> Box<dyn Iterator<Item = KV> + 'b> {
         range_with_prefix(self.storage, &self.prefix, start, end, order)
     }
+
+    fn set(&mut self, _key: &[u8], _value: &[u8]) {
+        unimplemented!("Use PrefixedStorage for writing")
```

Comment:
Ah, that's a way around it... fail runtime not compile time. Maybe less fighting the compiler and more docstrings would be useful.

I will let @webmaster128 decide on merging it, but nice solution.

Reply:
And I guess required for the erc20 contract?

Reply:
Yes. There are some helper functions that passed `ReadonlyStorage`. When changed their parameter to `dyn Storage`, this came out.